### PR TITLE
MA-1401: Generate alerts / tickets when account regisration fails with ESB / IT

### PR DESCRIPTION
**Pull request by chandramishra**
**Purpose:** Send alert mail when account registration with billing system has been failed
For more details Please refer this link : https://mountdiablo.jira.com/browse/MA-1401?filter=-1
**Changes:**  API and aws-services-onboarding & aws-services-organizations micro services
**Integration testing/functional testing:**
1) Create account when billing system not reachable
2) Create account with wrong api & secret key
3) Create account with existing email id
‌
**Unit Testing**
Not Available

### CODE REVIEW

**Review by RajkumarShivage — APPROVED** [review]
LGTM

File: src/index_enable_billing_activation.js [hunk 1]
```diff
@@ -1,55 +1,106 @@
...
     customerAwsId: accId,
     customerAwsName: params.name,
-    type: params.type ,
+    type: params.type,
     awsDesc: params.desc,
     activationDate: datetime
-  }
+  };
+  const billingUrl = `${params.billingUrl}?client_id=${params.apiKey}&client_secret=${params.secretKey}`;
+  console.log(billingUrl);
```
Comment (RajkumarShivage) [inline src/index_enable_billing_activation.js]: We should not print or log the access and secret key. Please remove access and secret key from console log.

File: src/index_enable_billing_activation.js [hunk 2]
```diff
@@ -1,55 +1,106 @@
...
   console.log(bodyjson);
```
Comment (RajkumarShivage) [inline src/index_enable_billing_activation.js]: It is not recommended to print or log the access and secret key.

File: src/index_enable_billing_activation.js [hunk 3]
```diff
@@ -1,55 +1,106 @@
...
+      // this is result of create account
+      accountId = retDoc.CreateAccountStatus.AccountId;
+    }
+    if (accountId) {
+      // sending request to billing system(rest call) for register account
+      registerAccount(accountId, billingInfo);
+    } else {
+      console.log(`customer account id not found: ${JSON.stringify(retDoc)}`);
+      const errmsg = `Found Error While Creating Account:  ${JSON.stringify(retDoc.CreateAccountStatus.FailureReason)}`;
+      sendAlertMailBySes(billingInfo, errmsg, billingInfo);
```
Comment (RajkumarShivage) [inline src/index_enable_billing_activation.js]: Is there a requirement to send an email if account creation fails?

File: src/index_enable_billing_activation.js [hunk 4]
```diff
@@ -1,55 +1,106 @@
...
+    }
+  } else {
+    console.log('customer account id not found');
+    sendAlertMailBySes(billingInfo, 'Found Error While Creating Account');
```
Comment (RajkumarShivage) [inline src/index_enable_billing_activation.js]: Is there a requirement to send an email if account creation fails?

**Review by RajkumarShivage — CHANGES_REQUESTED** [review]

File: src/index_enable_billing_activation.js (hunk 3 above)
Comment (chandramishra) [inline src/index_enable_billing_activation.js]: no any requirement . I did additional just for tracking but any way i can comment code for send alert mail

**Review by RajkumarShivage — APPROVED** [review]

**Review by SonalOjha — APPROVED** [review]
Looks fine

**Pull request merged**